lib/tests/test_mut_repo.rs

```diff
@@ -676,3 +676,75 @@ fn test_remove_wc_commit_previous_discardable() {
     mut_repo.rebase_descendants(&settings).unwrap();
     assert!(!mut_repo.view().heads().contains(old_wc_commit.id()));
 }
+
+#[test]
+fn test_reparent_descendants() {
```

Comment:
nit: squash the test into the parent commit (see https://martinvonz.github.io/jj/latest/contributing/#code-reviews)